src/azure-cli/azure/cli/command_modules/acs/custom.py

```diff
@@ -2619,6 +2620,9 @@ def _handle_addons_args(cmd, addons_str, subscription_id, resource_group_name, a
     # error out if '--enable-addons=monitoring' isn't set but workspace_resource_id is
     elif workspace_resource_id:
         raise CLIError('"--workspace-resource-id" requires "--enable-addons monitoring".')
+    if 'azure-policy' in addons:
+        addon_profiles['azurepolicy'] = ManagedClusterAddonProfile(enabled=True)
+        addons.remove('azure-policy')
```

Comment:
Musing thought (no change needed): I am worried that since we remove the addons and then check for remaining addons that we could easily run into problems if any dev forgets the addons.remove() line. But no need to refactor it in this PR

Reply:
yeah maybe something we need to handle later.